actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -412,36 +412,51 @@ def radio_button_tag(name, value, checked = false, options = {})
+      # * <tt>:disable_with</tt> - Defaults to value attribute. Value of this parameter will be used as the value for a
       #   disabled version of the submit button when the form is submitted. This feature is
-      #   provided by the unobtrusive JavaScript driver.
+      #   provided by the unobtrusive JavaScript driver. To disable this feature for a single submit tag
+      #   pass <tt>:data => { :disable_with => false }</tt>
       #
       # ==== Examples
       #   submit_tag
-      #   # => <input name="commit" type="submit" value="Save changes" />
+      #   # => <input name="commit" data-disable-with="Save changes" type="submit" value="Save changes" />
       #
       #   submit_tag "Edit this article"
-      #   # => <input name="commit" type="submit" value="Edit this article" />
+      #   # => <input name="commit" data-disable-with="Edit this article" type="submit" value="Edit this article" />
       #
       #   submit_tag "Save edits", disabled: true
-      #   # => <input disabled="disabled" name="commit" type="submit" value="Save edits" />
+      #   # => <input disabled="disabled" name="commit" data-disable-with="Save edits" type="submit" value="Save edits" />
       #
-      #   submit_tag "Complete sale", data: { disable_with: "Please wait..." }
-      #   # => <input name="commit" data-disable-with="Please wait..." type="submit" value="Complete sale" />
+      #   submit_tag "Complete sale", data: { disable_with: "Submitting..." }
+      #   # => <input name="commit" data-disable-with="Submitting..." type="submit" value="Complete sale" />
       #
       #   submit_tag nil, class: "form_submit"
       #   # => <input class="form_submit" name="commit" type="submit" />
       #
       #   submit_tag "Edit", class: "edit_button"
-      #   # => <input class="edit_button" name="commit" type="submit" value="Edit" />
+      #   # => <input class="edit_button" data-disable-with="Edit" name="commit" type="submit" value="Edit" />
       #
       #   submit_tag "Save", data: { confirm: "Are you sure?" }
-      #   # => <input name='commit' type='submit' value='Save' data-confirm="Are you sure?" />
+      #   # => <input name='commit' type='submit' value='Save' data-disable-with="Save" data-confirm="Are you sure?" />
       #
       def submit_tag(value = "Save changes", options = {})
         options = options.stringify_keys
+        tag_options = { "type" => "submit", "name" => "commit", "value" => value }.update(options)
 
-        tag :input, { "type" => "submit", "name" => "commit", "value" => value }.update(options)
+        if ActionView::Base.automatically_disable_submit_tag
+          disable_with_text = value.clone
+          disable_with_text = tag_options["data"][:disable_with] if tag_options["data"] && tag_options["data"].has_key?(:disable_with)
+          disable_with_text = tag_options["data-disable-with"] if tag_options.has_key?("data-disable-with")
```

Comment:
Then the variable name feels a little off :wink: And even so, there should be a simpler way to write the above.